### PR TITLE
Documentation & Code cleanup

### PR DESCRIPTION
**Pull request by neoblizz**

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -22,6 +22,8 @@ OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 SOFTWARE.
 -->
 
+**[README](../README.md)** » **Contributing to Iris**
```
Comment (mawad-amd) [inline docs/CONTRIBUTING.md:25]: this is neat.

File: images/iris-model.png
Comment (mawad-amd) [inline images/iris-model.png:1]: While this looks neat on dark mode, it doesn't look so good on light mode. Can we change white lines/text to something legible on white background (any other solution works too)?

Image

File: images/heap.png
Comment (mawad-amd) [inline images/heap.png:1]: Same comment on legibility of the "Virtual Address Space" text in light mode.

File: README.md [hunk 2]
```diff
@@ -23,41 +26,53 @@ Iris is a Triton-based framework for Remote Memory Access (RMA) operations. Iris
...
+### Iris Imports
+
 ```python
 import torch
 import triton
 import triton.language as tl
 import iris
+```
 
+### Device-side APIs
```
Comment (mawad-amd) [inline README.md]: I do like the idea of using sections, but I’d prefer to keep the code as a single block like before. Maybe we could add section titles as comments within the code instead? That way, I can still tell people they can click the copy icon at the top right of the snippet, paste it into a Python file, and it’ll just work. Splitting it into three separate snippets makes that harder to do.

File: README.md [hunk 3]
```diff
@@ -94,7 +111,7 @@ Check out our [examples](examples/) directory for ready-to-run scripts and usage
 
 Iris currently supports:
 
-- MI300X
+- MI300X, MI350X & MI355X
```
Comment (mawad-amd) [inline README.md:107]: Nice!

File: README.md [hunk 4]
```diff
@@ -23,41 +26,53 @@ Iris is a Triton-based framework for Remote Memory Access (RMA) operations. Iris
 
 1. [Peer-to-Peer Communication](examples/README.md)
 2. [Fine-grained GEMM & Communication Overlap](./docs/FINEGRAINED_OVERLAP.md)
+3. [Setup Alternatives](docs/SETUP_ALTERNATIVES.md)
```
Comment (mawad-amd) [inline README.md]: Can we add the [Programming Model](docs/PROGRAMMING_MODEL.md) bulletpoint

File: docs/PROGRAMMING_MODEL.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+-->
+
+**[README](../README.md)** » **Programming Model**
+
+# Programming Model
+
+<div style="overflow: hidden;">
+  <p style="float: left; width: 50%;">
+    Iris is an open-source triton-based framework for Remote Memory Access (RMA†) operations written in only a few 100 lines of code. Iris provides SHMEM-like APIs within Triton for Multi-GPU programming.
```
Comment (mawad-amd) [inline docs/PROGRAMMING_MODEL.md]: Footnote for † is missing. We can either add a footnote or just remove the dagger. It's written down below -- we can remove it.

File: docs/PROGRAMMING_MODEL.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+4. **Scalable by Design**
+    - Full scale-up (multi-GPU node) support
+    - Scale-out (multi-node) in progress
+
+
+## Simple `load` & `store` APIs
+
+```python
+@triton.jit
+def load(local_ptr, local_rank, remote_rank, heap_bases, mask=None):
```
Comment (mawad-amd) [inline docs/PROGRAMMING_MODEL.md]: I know these are copied from the slides but can we use the new code with current naming here instead?

**Review by mawad-amd — COMMENTED** [review]
Thanks for preparing this and for the cool logo. These changes are looking great but I left a few comments for further improvements.

File: docs/PROGRAMMING_MODEL.md [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+Copyright (c) 2025 Advanced Micro Devices, Inc. All rights reserved.
+-->
+
+**[README](../README.md)** » **Programming Model**
+
+# Programming Model
+
+<div style="overflow: hidden;">
+  <p style="float: left; width: 50%;">
+    Iris is an open-source triton-based framework for Remote Memory Access (RMA[^1]) operations written in only a few 100 lines of code. Iris provides SHMEM-like APIs within Triton for Multi-GPU programming.
```
Comment (mawad-amd) [inline docs/PROGRAMMING_MODEL.md]: Footnote not rendering correctly. Maybe need a space before the bracket?

Image

File: docs/PROGRAMMING_MODEL.md [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+Allocation routine for symmetric variables must be collective or offset must be known. Must all_gather the base heap addresses across all processes.
+
+```python
+@triton.jit
+def load(pointer, to_rank, from_rank, heap_bases, mask=None):
+    translated_ptr = __translate(pointer, from_rank, to_rank, heap_bases)
+    result = tl.load(translated_ptr, mask=mask)
+    return result
+
+@triton.jit
```
Comment (mawad-amd) [inline docs/PROGRAMMING_MODEL.md:110]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -21,43 +24,49 @@ Iris is a Triton-based framework for Remote Memory Access (RMA) operations. Iris
...
-2. [Fine-grained GEMM & Communication Overlap](./docs/FINEGRAINED_OVERLAP.md)
+1. [Programming Model](docs/PROGRAMMING_MODEL.md)
+2. [Peer-to-Peer Communication](examples/README.md)
+3. [Fine-grained GEMM & Communication Overlap](docs/FINEGRAINED_OVERLAP.md)
+4. [Setup Alternatives](docs/SETUP_ALTERNATIVES.md)
 
 ## API Example
 
-Iris matches PyTorch APIs on the host side and Triton APIs on the device side:
+### Iris Imports
```
Comment (mawad-amd) [inline README.md]:
```suggestion
```
Or move it inside the ```python block.

**Review by mawad-amd — APPROVED** [review]
Couple of minor comments but otherwise looks great. Thank you!

**Comment by neoblizz** [comment]
Done, merging now!
